tests/PHPStan/Parser/PhpDocParserTest.php

```diff
@@ -3046,6 +3046,30 @@ public function provideRealWorldExampleData(): \Iterator
 				),
 			]),
 		];
+
+		yield [
```

Comment:
Please add these tests that check for:

1) That trailing comma after last element is allowed
2) That leading comma at the beginning of the line is also allowed (I know it's ugly but I've seen it in the wild).

Thank you, otherwise 👍

Reply:
Ah, sure thing. Will do 👍

Reply:
Just a quick question, trailing comma should not work at all am I right?
`array<A, B,>` is not a correct notation

Where a the leading one should be fine, as long as there are two types given
```
array<
   A
   , B
>
```

Reply:
(Kotlin allows trailing comma in generic declarations since 1.4).

Reply:
Trailing commas are getting popular and even supported by PHP itself. Since this works: https://phpstan.org/r/e1c07184-7f6d-48ba-99cd-29dd41fe5349, I'm also fine if it will work in generics.

Reply:
We shouldn't support multiple trailing commas: `A,B,,` - that should be an error, otherwise they're fine.

Reply:
Thanks for informations. Will add support for them then :)